### ISSUE TITLE
[rush] `rush install --variant <variant-name>` doesn't work for the latest rush versions

### ISSUE THREAD

**Issue by vhuseinova-msft** [microsoft/rushstack#4784]
## Summary
The latest versions of rush (after 5.112.2) doesn't support variants anymore for `rush install` command even though an error (when running `rush update`) states that variants won't be supported in the next **major** release.
![Screenshot 2024-06-13 at 8 55 08 AM](https://github.com/microsoft/rushstack/assets/98852890/0df46aaf-00b3-4f82-ade5-3a6ba7fea9df)

![Screenshot 2024-06-13 at 8 48 26 AM](https://github.com/microsoft/rushstack/assets/98852890/0c70125a-8fe9-45f4-a395-11fc0b9259a8)

## Repro steps
- open a monorepo with variants support
- install the latest version of rush
- try to run `rush install --variant <variant-name>`

  **Expected result:** the variants ae supported until rush version 6

  **Actual result:** the variants aren't supported anymore

## Standard questions

Please answer these questions to help us investigate your issue more quickly:

| Question | Answer |
| -------- | -------- |
| `@microsoft/rush` globally installed version? | 5.127.0 |
| `rushVersion` from rush.json? | 5.128.0 |
| `useWorkspaces` from rush.json? | not specified |
| Operating system? | Mac |
| Would you consider contributing a PR? | No |
| Node.js version (`node -v`)? | v20.14.0 |

**Comment by iclanton** [comment]
Yeah the variants feature was removed in favor of subspaces. That error message is saying that Rush won't give a hard error if the `"variants"` field in `rush.json` is present until Rush 6, but the feature has been removed.

@g-chao - can you provide guidance on how @vhuseinova-msft can use subspaces?

**Comment by kornza12** [comment]
.
